h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -371,13 +371,17 @@ void checkAndInheritModelProperties() {
         // !beenHere: this is the first base_model
         this.modelCategory = aModel._output.getModelCategory();
         this._dist = new Distribution(distributionFamily(aModel));
-        _output._domains = Arrays.copyOf(aModel._output._domains, aModel._output._domains.length);
-
         // TODO: set _parms._train to aModel._parms.train()
 
-        _output.setNames(aModel._output._names, aModel._output._column_types);
-        this.names = new NonBlockingHashSet<>();
-        this.names.addAll(Arrays.asList(aModel._output._names));
```

Comment:
it seems better to rely on `train` domains/names than on the output domains provided by the first base model.
If this one encoded the categoricals, then the `output.domains` keeps only the target, as there's no categoricals anymore...